lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -614,20 +614,19 @@ def initial_manifest?(path, manifest_setting)
     string_path == manifest_setting || string_path.start_with?(manifest_setting)
   end
 
-  def get_module_relative_path(file_path, modulepath_directories)
-    clean_file = file_path.cleanpath
-    parent_path = modulepath_directories.find { |path_dir| is_parent_dir_of(path_dir, clean_file) }
+  # Get the path of +file_path+ relative to the first directory in
+  # +clean_modulepath_names+ that is an ancestor of +file_path+. Return NO_PATH
+  # if none is found.
+  def get_module_relative_path(file_path, clean_modulepath_names)
+    clean_file = file_path.cleanpath.to_s
+    parent_path = clean_modulepath_names.find do |pathname|
+	    clean_file.start_with?(pathname.to_s + File::SEPARATOR)
```

Comment:
This may not work with case-insensitive file names. Need to call `Pathname#ascend` and check if the parent equals `pathname`